lib/tapyrus/tip0137.rb

```diff
@@ -0,0 +1,131 @@
+module Tapyrus
+  module TIP0137
+    # @param key [Tapyrus::Key] private key
+    # @param txid [String] txid
+    # @param index [Integer] index of transaction output
+    # @param color_id [Tapyrus::Color::ColorIdentifier] a valid color id
+    # @param value [Integer] non-negative integer less than 2^64
+    # @param script_pubkey [Tapyrus::Script] script pubkey in transaction output
+    # @param address [String] Tapyrus address for transaction output
+    # @param data [String] hexdecimal string
+    # @param client [Tapyrus::RPC::TapyrusCoreClient] rpc client
+    def sign_message(
+      key,
+      txid: nil,
+      index: nil,
+      color_id: nil,
+      value: nil,
+      script_pubkey: nil,
+      address: nil,
+      data: nil,
+      client: nil
+    )
+      validate_payload!(
+        txid: txid,
+        index: index,
+        color_id: color_id,
+        value: value,
+        script_pubkey: script_pubkey,
+        address: address,
+        data: data
+      )
+      if client
+        validate_on_blockchain!(
+          client: client,
+          txid: txid,
+          index: index,
+          color_id: color_id,
+          value: value,
+          script_pubkey: script_pubkey
+        )
+      end
+      data = {
+        txid: txid,
+        index: index,
+        color_id: color_id&.to_hex,
+        value: value,
+        script_pubkey: script_pubkey&.to_hex,
+        address: address,
+        data: data
+      }
+      JWS.encode(data, key.priv_key)
+    end
+
+    #   # @param jws [String] JWT Web Token
+    #   # @param key [Tapyrus::Key] public key
+    #   # @param client [Tapyrus::RPC::TapyrusCoreClient] rpc client
+    #   # @return decoded JSON Object
+    def verify_message(jws, key, client: nil)
```

Comment:
In addition to this method, I think it would be nice to have a wrapper that validates the validity and returns the result as a Boolean.